### PR TITLE
Add Skip to DSL to allow skipping at runtime

### PR DESCRIPTION
**Pull request by karlkfi**
Example Usage:

```
It("handles skips", func() {
	Skip("show this message and location to the user")
	println("NEVER SEE THIS")
})
```

Implements feature request: #168

### CODE REVIEW

File: internal/failer/failer.go [hunk 1]
```diff
@@ -77,3 +77,16 @@ func (f *Failer) Drain(componentType types.SpecComponentType, componentIndex int
...
 	return failure, outcome
 }
+
+func (f *Failer) Skip(message string, location types.CodeLocation) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	if f.state == types.SpecStatePassed {
+		f.state = types.SpecStateSkipped
+		f.failure = types.SpecFailure{
```
Comment (karlkfi) [inline internal/failer/failer.go]: This could be something other than a SpecFailure if there were an interface with Message() and Location()

**Comment by karlkfi** [comment]
@onsi Had a chance to look at this? What's the verdict?

**Comment by onsi** [comment]
Sorry for the delay @karlkfi - this looks fine to me.  Will merge it in.

**Comment by karlkfi** [comment]
:tada:

**Pull request merged**